//repeated， required， optional

syntax = "proto2";

//协议外壳
message MainProto{
	required int32 playerID = 1;
	required int32 messageID = 2;
	optional string messageName = 3;
	optional string messageData = 4;
	optional double serverTime = 5;
}


//获取信息
message MessagePeopleReq{
	optional int32 peopleID = 1;
	optional bool isChina = 2;
}
//返回信息
message MessagePeopleRsp{
	optional string name = 1;
	optional string phone = 2;
	optional int32 age = 3;
}


//请求登陆
message MessageLoginReq{
	
}
message MessageLoginRsp{
	optional int32 playerID = 1;
}

//请求开始游戏
message MessageStartGameReq{
}
message MessageStartGameRsp{
	
}

//请求发牌
message MessageDealReq{
	
}
message MessageDealRsp{
	optional int32 upNum = 1; //上家牌的数量
	optional int32 downNum = 2; //下家牌的数量
	repeated MessageDataPoker pokerList = 3;
}


//同步状态
message MessageUpdateStateReq{
	
}
message MessageUpdateStateRsp{
	optional int32 playerIdx = 1; 	//可以操作的玩家位置, (0.自己, 1.下家, 2.上家)
	optional int32 stateType = 2;   //操作类型，0没开始 1抢地主，2出牌
	repeated MessageDataPoker midPoker = 3;	//牌桌上的牌
	repeated MessageDataPoker landlordPoker = 4; //地主牌
	repeated MessageDataPoker selfPoker = 5;	//自己的牌
	optional int32 upPokerNum = 6;				//上家牌数
	optional int32 downPokerNum = 7;			//下家家牌数
	optional int32 laveTime = 8;		//剩余时间
}


//抢地主
message MessageGradLandlordReq{
	optional bool isGrad = 1;	//是否抢地主
}
message MessageGradLandlordRsp{
	
}

//出牌
message MessageOutPokerReq{
	repeated MessageDataPoker poker = 1;
}
message MessageOutPokerRsp{
}

//////////////////////////////////////
message MessageDataPoker{
	optional int32 color = 1;		//颜色(1:黑,2:红,3:梅,4.方,5.大王,6.小王)
	optional int32 number = 2;	//数字(16:王, 14:A, 15:2, 3:3, 4:4...j:11, Q:12, K:13)
}